jcl/src/java.base/share/classes/java/lang/StringBuilder.java

```diff
@@ -250,47 +258,47 @@ public StringBuilder append (char chars[], int start, int length) {
 	if (start >= 0 && 0 <= length && length <= chars.length - start) {
 		int currentLength = lengthInternal();
 		int currentCapacity = capacityInternal();
-		
+
 		int newLength = currentLength + length;
 		if (newLength < 0) {
 			/*[MSG "K0D01", "Array capacity exceeded"]*/
 			throw new OutOfMemoryError(com.ibm.oti.util.Msg.getString("K0D01")); //$NON-NLS-1$
 		}
-		
+
 		if (String.COMPACT_STRINGS) {
 			// Check if the StringBuilder is compressed
 			if (count >= 0 && String.canEncodeAsLatin1(chars, start, length)) {
 				if (newLength > currentCapacity) {
 					ensureCapacityImpl(newLength);
 				}
-				
+
 				String.compress(chars, start, value, currentLength, length);
-				
+
 				count = newLength;
 			} else {
 				// Check if the StringBuilder is compressed
 				if (count >= 0) {
-					currentCapacity = decompress(newLength);
+					currentCapacity = decompress(newLength, currentLength);
```

Comment:
The pattern that occurs here and several other places:
```
			if (count >= 0) {
				currentCapacity = decompress(newLength, currentLength);
			}
			if (newLength > currentCapacity) {
				ensureCapacityImpl(newLength);
			}
```
can be simplified to:
```
			if (count >= 0) {
				decompress(newLength, currentLength);
			} else if (newLength > currentCapacity) {
				ensureCapacityImpl(newLength);
			}
```
eliminating the need for `decompress()` to return the (possibly updated) capacity.

Reply:
Changed this pattern as suggested